server/src/main/java/com/objectcomputing/checkins/services/settings/SettingsServicesImpl.java

```diff
@@ -1,53 +1,51 @@
 package com.objectcomputing.checkins.services.settings;
 
-import com.objectcomputing.checkins.exceptions.AlreadyExistsException;
 import com.objectcomputing.checkins.exceptions.BadArgException;
 import com.objectcomputing.checkins.exceptions.NotFoundException;
-import com.objectcomputing.checkins.services.memberprofile.currentuser.CurrentUserServices;
+import io.micronaut.core.annotation.Nullable;
 import jakarta.inject.Singleton;
 import jakarta.validation.constraints.NotNull;
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Optional;
 import java.util.UUID;
 import java.util.stream.Collectors;
 
 @Singleton
 public class SettingsServicesImpl implements SettingsServices {
 
     private final SettingsRepository settingsRepository;
-    private final CurrentUserServices currentUserServices;
 
-    public SettingsServicesImpl(SettingsRepository settingsRepository, CurrentUserServices currentUserServices) {
+    public SettingsServicesImpl(SettingsRepository settingsRepository) {
         this.settingsRepository = settingsRepository;
-        this.currentUserServices = currentUserServices;
     }
 
     public Setting save(Setting setting) {
         if (setting.getId() != null) {
-            throw new AlreadyExistsException("This setting already exists");
+            throw new BadArgException("Setting ID is autogenerated by the server upon creation, and should not be provided.");
         }
+        validateSettingOption(setting);
         return settingsRepository.save(setting);
     }
 
     public Setting update(Setting setting) {
-        if (setting.getId() != null && settingsRepository.findById(setting.getId()).isPresent()) {
+        validateSettingOption(setting);
+        if ((settingsRepository.existsByIdAndName(setting.getId(), setting.getName()))) {
             return settingsRepository.update(setting);
         } else {
             throw new BadArgException(String.format("Setting %s does not exist, cannot update", setting.getId()));
         }
     }
 
-    public List<SettingsResponseDTO> findByName(String name) {
-        List<Setting> searchResult = settingsRepository.findByUserId(currentUserServices.getCurrentUser().getId());
+    public List<SettingsResponseDTO> findByName(@Nullable String name) {
```

Comment:
I don't know how large of a dataset we expect in this table, but I wonder if it would make sense to test for `name != null` and then either query all the settings or only those with a matching name, based on that result?